packages/vitest/src/runtime/suite.ts

```diff
@@ -14,7 +14,10 @@ export const test = createTest(
   },
 )
 
-function formatTitle(template: string, items: any[]) {
+function formatTitle(template: string, items: any[], idx: number) {
+  // '%#' match index of the test case
+  template = template.replace(/\%\#/g, `${idx}`)
```

Comment:
Thinking about it, we should check that the `%` isn't scaped here, no? To account for the case: `%%#`

Maybe it should be something like:
```
template.replace(/%%/g,`__vitest_escaped_%__`).replace(/\%\#/g, `${idx}`).replace(/__vitest_escaped_%__/g, '%%')
```

We could still merge this PR and review the escaping of `%` in general in a future PR, looking at the code below these edge cases may not be considered for other cases.

Reply:
Oh, It's a detail what I ignored. 
Thanks for letting me learn something.

Reply:
A nitpick, but we may also want to avoid doing this dance if not needed. So we could wrap it with `if (template.includes('%#')) {` so we only do it when there is a chance the replacement will be applied.

Reply:
I thought about this before, At the time I thought `includes` and `replace` both need traverse the template that add a layer of judgment is redunant.

but now I think what you're saying makes sense.